lib/_tls_wrap.js

```diff
@@ -1138,8 +1140,17 @@ exports.connect = function(...args /* [port,] [host,] [options,] [cb] */) {
   if (options.session)
     socket.setSession(options.session);
 
-  if (options.servername)
+  if (options.servername) {
+    if (ipServernameWarned === false && net.isIP(options.servername)) {
+      process.emitWarning(
+        'Setting the TLS ServerName to an IP address is not supported by ' +
+        'RFC6066. This will be ignored in a future version.',
+        'UnsupportedWarning'
```

Comment:
Can you emit `'DeprecationWarning'` plus a DEPxxxx code here?  The code needs to be added to `doc/api/deprecations.md`.

Reply:
Just a note, making this a DeprecationWarning would make this semver-major

Reply:
I'm a little out of my depth here with regards to both node's deprecation and its versioning semantics.
Depending on what level the "drop hostname on IP" thing ultimately ends up at (could be here; could be tls.js?), I'm not sure this makes sense to consider a "deprecation" per se? Most if not all existing external users should see the warning for a version, then see it silently go away with no change necessary in a future version.

Reply:
I think the sentence `This will be ignored in a future version.` implies that it is a deprecation.  Making it a `DeprecationWarning` gives us a proper documentation for the behavior change.

Also this reminds me, can you add a entry to the `changes` YAML field in the documentation of `tls.connect`?